Change completer width styling

ryantberg opened the pull request:
Fixes kiteco/kiteco#11181

Reduces the variance in completer width as you type, and also makes completion lists with only short completions feel less cramped.

These style changes should be upstreamed to jupyterlab core, but we can patch them in the extension for now.

plungg commented:
@ryantberg 
* Let's add a 5 char padding between completion and hint. If no hint, no need for padding.
* Let’s do a max width of 42 characters, which is what Kite in VS Code does. This 42 character width includes the completion, hint, and pro star. 
* If a completion is too long, add an ellipsis at the end. 
![image](https://user-images.githubusercontent.com/5544312/83069115-35c83e00-a01e-11ea-9191-9e6bfa615191.png)

its-dhung commented:
> Let’s do a max width of 42 characters, which is what Kite in VS Code does. This 42 character width includes the completion, hint, and pro star.

Assume most users use a 12pt font, what does this translate into pixels?

plungg commented:
Updated with the pixel width.

ryantberg commented:
@plungg I've updated this. Unfortunately, there's no straightforward way to add a maximum width to the entire type-able space, but I've added a maximum width to the completion text, with the ellipsis cutoff as specified. I picked a max-width that roughly corresponds to the type-able area being 430px wide.

I've also fixed a couple of other minor issues that cause the completer width to fluctuate, so it should be much less distracting. Let me know if this works for you, and if anything else sticks out, styling-wise.

plungg commented:
@ryantberg everything looks great, checked out the min, max, overflow. 

Couldn't test `Settings > Increase Text Editor Font Size` but it seems broken in regular JupyterLab as well. I remember using that when I first tested this out. 🤷‍♂️
